cdk_framework/EKS/lib/utils/validate-cluster-config.ts

```diff
@@ -0,0 +1,146 @@
+        }
+        const val = Object(value)
+        if(clusterNamesSet.has(key)){
+            throw new Error('Cannot have multiple clusters with the samne name')
+        }
+        clusterNamesSet.add(key)
+        validateRequiredFields(val)
+        for(const [k, v] of Object.entries(val)){
+            if(!supportedFields.has(k)){
+                throw new Error('Incompatible field type')
+            }
+            switch(k){
+                case 'version':
+                    val[k] = validateVersion(String(v))
+                    break;
+                case 'launch_type':
+                    val[k] = convertAndValidateLaunchType(val)
+                    break
+            }
+        }
+    }
+}
+
+function validateRequiredFields(fields: any){
+    for(const field of requiredFields){
+        if(!fields[field]){
+            throw new Error('Required field - ' + field + ' - not provided')
+        }
+    }
+}
+
+function checkToSetUpDefaults(fields: any){
+    if(!fields['ec2_instance']){
+        fields['ec2_instance'] = 'm5'
+    }
+    if(!fields['node_size']){
+        fields['node_size'] = 'large'
+    }
+}
+
+function validateVersion(version: string){
+    
+    if(version === '1.2'){
+        version = '1.20'
+    }
+
+    if(!supportedVersions.has(version)){
+        throw new Error('Version needs to be a value of one of the following: ' + Array.from(supportedVersions).join(', '));
+    }
+    return version
+}
+
+function convertAndValidateEC2Instance(instance: string){
+    if(instance === null || !instance || instance == 'null'){
+        throw new Error('EC2 instance type was not provided for ec2 cluster')
+    }
+    const adjustedType = instance.toLowerCase()
+    if(!supportedCPUArchitectures.has(adjustedType)){
+        throw new Error('Improper instance type or provided faulty ec2_instance/node_size for fargate cluster')
+    }
+    return adjustedType
+}
+
+function convertAndValidateLaunchType(type: any){
+    const launchType = Object(type['launch_type'])
+    if(Object.keys(launchType).length != 1){
+        throw new Error('More than 1 launch_type provided or none provided')
+    }
+    if(launchType['fargate'] !== undefined){
+        return launchType
+    } else if(launchType['ec2'] !== undefined){
+        const launchData = Object(launchType['ec2'])
+        checkToSetUpDefaults(launchData)
+        for(const [k, v] of Object.entries(launchData)){
+            if(!supportedFields.has(k)){
+                throw new Error('Provided field type, ' + k + ', is not a compatible field type')
+            }
+            switch(k){
+                case 'ec2_instance':
+                    launchData[k] = convertAndValidateEC2Instance(String(v))
+                    break
+                case 'node_size':
+                    launchData[k] = validateNodeSize(String(v), String(launchData['ec2_instance']))
+                    break;
+            }
```

Comment:
Instead of checking the field type should we instead have a `default` case that will throw an error if reached?